src/main.qml

```diff
@@ -17,27 +17,205 @@
+            ctx.stroke()
+            ctx.beginPath()
+            ctx.arc(x, y, (app.width / 2.9), gap2, end, false)
+            ctx.stroke()
+        }
     }
 
     Label {
         id: bpmText
         anchors.centerIn: parent
         horizontalAlignment: Text.AlignHCenter
-       //% "Measuring..."
-        text: app.bpm > 0 ? app.bpm : qsTrId("id-measuring")
+        font.letterSpacing: app.bpm > 0 ?
+                                -app.width * 0.008 :
+                                0
+        font.pixelSize: app.bpm > 0 ?
+                            app.bpm / 100 >= 1 ?
+                                app.height*0.29 :
+                                app.height*0.33 :
+                            app.height*0.06
+        text: app.bpm > 0 ?
+                  app.bpm :
+                  qsTrId("id-measuring")
+        Timer {
+            id: timer
+            interval: 800
+            running: true
+            repeat: app.bpm === 0
+            onTriggered: app.bpm === 0 ?
+                             bpmText.opacity = bpmText.opacity === 0 ?
+                                 1 :
+                                 0 :
+                             bpmText.opacity = 1
+        }
+        Behavior on opacity { NumberAnimation { duration: 600; easing.type: Easing.OutQuad } }
+
+    }
+
+    Text {
+        id: lastBpmText
+        z: 2
+        color: "#ffffff"
+        anchors {
+            horizontalCenter: app.horizontalCenter
+            horizontalCenterOffset: -app.width * 0.345
+            verticalCenter: app.verticalCenter
+        }
+        font.letterSpacing: -heartPicture.width * 0.0018
+        font.pixelSize: app.height*0.086
+        font.styleName: "Bold"
+        text: app.lastBpm > 0 ?
+                  app.lastBpm :
+                  ""
+        Text {
+            id: arrowShape
+            z: 1
+            anchors{
+                left: lastBpmText.right
+                leftMargin: app.width * 0.02
+                verticalCenter: lastBpmText.verticalCenter
+            }
+            font.family: "Source Sans Pro"
+            font.pixelSize: app.height * 0.07
+            opacity: 1
+            color: "#6638FF12"
+            text: lastBpm > 0 ? "\u25B6" : ""
+        }
+        transform: Rotation {
+            origin.x: (app.width * 0.345) + (lastBpmText.width * 0.5)
+            origin.y: lastBpmText.height * 0.5
+            angle: lastBpmAngle
+            Behavior on angle {
+                NumberAnimation {
+                    duration: 1000
+                    easing.type: Easing.OutCirc
+                }
+            }
+        }
+    }
+
+    Text {
+        id: heartPicture
+        anchors{
+            horizontalCenter: app.horizontalCenter
+            verticalCenter: app.verticalCenter
+            verticalCenterOffset: app.height * 0.31
+        }
+        font.pixelSize: pulseWidth
+        opacity: 1
+        color: "red"
```

Comment:
No need to define \u2764 as red since it already is. Also `opacity: 1` is redundant.